cipher/stream-cipher/src/follower.rs

```diff
@@ -0,0 +1,475 @@
+        de_factory: DEF,
+        verifier_factory: VF,
+    ) -> DEStreamCipherFollower<C, DEF, DE, VF, V> {
+        let ctr_mode_config = CounterModeConfigBuilder::default()
+            .id(format!("{}/ctr", config.id))
+            .role(Role::Follower)
+            .start_ctr(config.start_ctr)
+            .concurrency(config.concurrency)
+            .build()
+            .expect("CounterMode config should be valid");
+
+        DEStreamCipherFollower {
+            config,
+            state: State::default(),
+            channel,
+            ctr_mode: CtrMode::new(ctr_mode_config, de_factory),
+            verifier_factory,
+            _cipher: PhantomData,
+            _verifier: PhantomData,
+        }
+    }
+
+    async fn verify_plaintext(
+        &mut self,
+        plaintext_labels: FullLabels,
+        key_stream_labels: FullLabels,
+        ciphertext: Vec<u8>,
+    ) -> Result<(), StreamCipherError> {
+        let circ = nbyte_xor(ciphertext.len());
+
+        let id = format!("{}/{}", self.config.id, self.state.execution_id);
+        self.state.execution_id += 1;
+
+        let verifier_config = VerifierConfigBuilder::default()
+            .id(id.clone())
+            .circ(circ.clone())
+            .build()
+            .expect("Verifier config should be valid");
+
+        let verifier = self.verifier_factory.create(id, verifier_config).await?;
+
+        let input_plaintext =
+            FullEncodedInput::from_labels(circ.inputs()[0].clone(), plaintext_labels)
+                .expect("Labels should be valid");
+
+        let input_key_stream =
+            FullEncodedInput::from_labels(circ.inputs()[1].clone(), key_stream_labels)
+                .expect("Labels should be valid");
+
+        let expected_output = circ.outputs()[0]
+            .clone()
+            .to_value(ciphertext)
+            .expect("Ciphertext should be valid");
+
+        let gen_labels = FullInputSet::new(vec![input_plaintext, input_key_stream])
+            .expect("Inputs should be valid");
+
+        _ = verifier
+            .verify_and_summarize(
+                gen_labels,
+                vec![],
+                vec![circ.inputs()[0].clone()],
+                vec![expected_output],
+            )
+            .await?;
+
+        Ok(())
+    }
+
+    async fn build_plaintext_labels(
+        &mut self,
+        len: usize,
+    ) -> Result<FullLabels, StreamCipherError> {
+        let encoder = self
+            .state
+            .encoder
+            .clone()
+            .ok_or(StreamCipherError::EncoderNotSet)?;
+        let mut encoder = encoder.lock().await;
+        let delta = encoder.get_delta();
+        let stream = encoder.get_stream(self.config.encoder_text_stream_id);
+        let labels = FullLabels::generate(stream, len * 8, Some(delta));
+        drop(encoder);
+
+        Ok(labels)
+    }
+
+    async fn build_ctr_share_labels(&mut self) -> Result<KeyBlockLabels, StreamCipherError> {
```

Comment:
this is an exact copy of the leader's `build_ctr_share_labels`. we should be able to factor it out into
a standalone fn 
build_ctr_share_labels(labels, encoder, encoder_default_stream_id)

On this note: shouldn't we pass  `encoder_default_stream_id` when instantiating the encoder?